tests/unit/anchore_engine/services/apiext/api/controllers/test_utils.py

```diff
@@ -312,30 +314,40 @@ def test_normalization_and_validation():
                 }
             },
         ),
-    ]
-
-    bad_requests = [
+        # Tag + Digest only (no ts)
+        (
+            {
+                "tag": "docker.io/library/nginx:latest",
+                "digest": test_digest,
+            },
+            {
+                "source": {
+                    "digest": {
+                        "pullstring": "docker.io/library/nginx@{}".format(test_digest),
+                        "tag": "docker.io/library/nginx:latest",
+                    }
+                }
+            },
+        ),
         # Malformed tag
         ({"tag": "docker.io/library/nginx@sha123"}, Exception),
-        # Tag + Digest only (no ts)
-        ({"tag": "docker.io/library/nginx:latest", "digest": "sh256:abc"}, Exception),
-        # Digest Only
-        ({"digest": "sh256:abc"}, Exception),
-        # Digest pullstring only
-        ({"digest": "docker.io/nginx@sha256:abc"}, Exception),
-    ]
-
-    matrix = good_requests + bad_requests
-
-    for test_input, result in matrix:
-        if type(result) == type and issubclass(result, Exception):
-            with pytest.raises(result):
-                normalized = api_utils.normalize_image_add_source(test_input)
-                api_utils.validate_image_add_source(normalized, api_spec)
-        else:
+        # Tag + Digest only (no ts) (malformed digest)
+        ({"tag": "docker.io/library/nginx:latest", "digest": "sh256:abc"}, BadRequest),
+        # Digest Only (malformed digest)
+        ({"digest": "sh256:abc"}, BadRequest),
```

Comment:
I must have been confused because the tag+digest test was failing with       `anchore_engine.apis.exceptions.BadRequest: {"message": "Must have digest-based pull string", "detail": "{\"invalid_value\": \"sh256:abc\"}", "httpcode": 400}`. But that test was meaningless, as the behavior changed with the fix for the issue. I've fixed the comments and the SHAs and removed that test case. Thanks!